File: scratch/update_versions.py
import os
import re

files = [
    r"c:\app2026\accountspro\src\App.jsx",
    r"c:\app2026\accountspro\src\10MAY\App.jsx",
    r"c:\app2026\accountspro\src\23MAY26\App.jsx",
    r"c:\app2026\accountspro\src\31may\App.jsx",
    r"c:\app2026\accountspro\src\ManagementDashboard.jsx",
    r"c:\app2026\accountspro\src\RegistersDashboard.jsx",
    r"c:\app2026\accountspro\src\ReportsV2.jsx",
    r"c:\app2026\accountspro\src\UserManualModal.jsx",
    r"c:\app2026\accountspro\scratch\api_tester.html"
]

for f in files:
    if os.path.exists(f):
        with open(f, 'r', encoding='utf-8') as file:
            content = file.read()
        
        # Replacements
        content = re.sub(r'2\.6\.2', '2.6.3', content)
        content = re.sub(r'2\.5\.1', '2.6.3', content)
        content = re.sub(r'2\.5\.3', '2.6.3', content)
        content = re.sub(r'v 2\.5\b', 'v 2.6.3', content)
        
        with open(f, 'w', encoding='utf-8') as file:
            file.write(content)

print("Versions updated.")

File: scratch/update_versions.js
const fs = require('fs');
const path = require('path');

// Update main AccountsPro files
const mainFiles = [
    'src/App.jsx',
    'src/main.jsx',
    'src/ManagementDashboard.jsx',
    'src/PackagingSmartReportModal.jsx',
    'src/RegistersDashboard.jsx',
    'src/ReportsV2.jsx',
    'src/UserManualModal.jsx',
    'package.json',
    'public/version.json'
];

mainFiles.forEach(file => {
    const filePath = path.join(__dirname, '..', file);
    if (!fs.existsSync(filePath)) {
        console.warn(`File not found: ${filePath}`);
        return;
    }
    let content = fs.readFileSync(filePath, 'utf8');

    // package.json: version
    if (file === 'package.json') {
        content = content.replace(/"version":\s*"[^"]+"/g, '"version": "2.6.7"');
    } else if (file === 'public/version.json') {
        content = content.replace(/"version":\s*"[^"]+"/g, '"version": "V2.6.7"');
    } else if (file === 'src/App.jsx') {
        // App.jsx replaces 2.6.3 with 2.6.7
        content = content.replace(/2\.6\.3\s*\(April\s*2026\)/g, '2.6.7 (June 2026)');
        content = content.replace(/v2\.6\.3/g, 'v2.6.7');
        content = content.replace(/v\s*2\.6\.3/g, 'v 2.6.7');
        content = content.replace(/"2\.6\.3"/g, '"2.6.7"');
    } else {
        // Replaces 2.6.6 with 2.6.7
        content = content.replace(/2\.6\.6/g, '2.6.7');
    }

    fs.writeFileSync(filePath, content);
    console.log(`Updated main app file: ${file}`);
});

// Update QuickAccPro files
const quickFiles = [
    'quickaccpro/package.json',
    'quickaccpro/src/components/Layout.jsx',
    'quickaccpro/src/components/ApiKeyLogin.jsx'
];

quickFiles.forEach(file => {
    const filePath = path.join(__dirname, '..', file);
    if (!fs.existsSync(filePath)) {
        console.warn(`File not found: ${filePath}`);
        return;
    }
    let content = fs.readFileSync(filePath, 'utf8');

    if (file === 'quickaccpro/package.json') {
        content = content.replace(/"version":\s*"[^"]+"/g, '"version": "1.4.0"');
    } else {
        // replace v1.3 with v1.4
        content = content.replace(/v1\.3/g, 'v1.4');
    }

    fs.writeFileSync(filePath, content);
    console.log(`Updated quick app file: ${file}`);
});

console.log('Version updates completed.');
